### ISSUE TITLE
[Ruby 2.5] Layout/RescueEnsureAlignment issue

### ISSUE THREAD

**Issue by Edouard-chin** [rubocop-hq/rubocop#6433]
👋 ,

Not sure if this is a rubocop bug or just the way it should be on 2.5 now that `begin` statement are superfluous inside block:

```ruby
something = foo do
  puts 'bar'
rescue => e
  puts e
end

# rescue not aligned with foo
```

Cop is passing with this syntax, but it's not really pretty .

```ruby
something = foo do
  puts 'bar'
            rescue => e
              puts e
end
```

**Comment by gmcgibbon** [comment]
I will take a look at this.

**Comment by richardmonette** [comment]
I'm running into this exact scenario here https://github.com/Shopify/shopify/pull/176657/files#diff-8a02f83394bcc83f212d8a8133b8cc4cR52

I think ideally we would not need to do the weird indents as this should be supported in a natural formatting (e.g. https://blog.bigbinary.com/2017/10/24/ruby-2.5-allows-rescue-inside-do-end-blocks.html)
